lib/dockerregistry/inventory.go

```diff
@@ -2527,42 +2527,121 @@ func GetDeleteCmd(
 		cmd)
 }
 
-// Contains checks whether a given Manifest mentions the contents of a
-// gcrPayload (re-interpreted as a FQIN or PQIN).
-// nolint[gocyclo]
-func (m Manifest) Contains(gcrPayload GCRPubSubPayload) bool {
-	fqinOnly := len(gcrPayload.Tag) == 0
+// GcrPayloadMatch holds bitwise flags for matching a GCRPubSubPayload against a
+// promoter manifest.
+//
+// For an example of how bitwise flags work, see
+// https://yourbasic.org/golang/bitmask-flag-set-clear/.
+type GcrPayloadMatch uint
+
+const (
+	// FlagPathMatched is set if the registry + image path (everything leading
+	// up to either the digest or a tag) matches.
+	FlagPathMatched GcrPayloadMatch = 1 << iota
+	// FlagDigestMatched is set if the digest in the payload matches a digest in
+	// the promoter manifest. This is ONLY matched if the path also matches.
+	FlagDigestMatched
+	// FlagTagMatched is ONLY matched if the digest also matches.
```

Comment:
I do think these rules suggest it isn't a bitset after all, hence why I suggested just returning `(bool, Info)`